src/game/console_cmd.c

```diff
@@ -213,6 +148,54 @@ static bool Console_Cmd_Teleport(const char *const args)
         }
     }
 
+    // Nearest item of this name
+    if (!String_Equivalent(args, "")) {
+        int32_t match_count = 0;
+        GAME_OBJECT_ID *matching_objs = Object_IdsFromName(args, &match_count);
+
+        const ITEM_INFO *best_item = NULL;
+        for (int i = 0; i < match_count; i++) {
+            const GAME_OBJECT_ID obj_id = matching_objs[i];
+
+            bool matched = false;
+            int32_t best_distance = INT32_MAX;
+            for (int16_t item_num = 0; item_num < Item_GetTotalCount();
+                 item_num++) {
+                const ITEM_INFO *const item = &g_Items[item_num];
+                if (item->object_number != obj_id
+                    || (item->flags & IF_KILLED_ITEM)) {
```

Comment:
If you `tp puzzle` in Lost Valley, pick it up, and then enter the same command again, it takes you to the same position. I think we'll need something like this - item status changes to `IS_INVISIBLE` on pickup, but also `IS_DEACTIVATED` if that item has a trigger (like in Valley here). This can maybe be cleaner though - removing the `is_pickup` check means you wouldn't teleport to enemies who are initialised invisible (like Larson in Qualopec).
```diff
diff --git a/src/game/console_cmd.c b/src/game/console_cmd.c
index c0eae8b8..db4e7c8e 100644
--- a/src/game/console_cmd.c
+++ b/src/game/console_cmd.c
@@ -156,6 +156,7 @@ static bool Console_Cmd_Teleport(const char *const args)
         const ITEM_INFO *best_item = NULL;
         for (int i = 0; i < match_count; i++) {
             const GAME_OBJECT_ID obj_id = matching_objs[i];
+            bool is_pickup = Object_IsObjectType(obj_id, g_PickupObjects);
 
             bool matched = false;
             int32_t best_distance = INT32_MAX;
@@ -163,7 +164,10 @@ static bool Console_Cmd_Teleport(const char *const args)
                  item_num++) {
                 const ITEM_INFO *const item = &g_Items[item_num];
                 if (item->object_number != obj_id
-                    || (item->flags & IF_KILLED_ITEM)) {
+                    || (item->flags & IF_KILLED_ITEM)
+                    || (is_pickup
+                        && (item->status == IS_INVISIBLE
+                            || item->status == IS_DEACTIVATED))) {
                     continue;
                 }
 ```

Reply:
Thanks for noticing this and suggesting the patch. Applied :)